tiledb/sm/filesystem/s3.cc

```diff
@@ -596,6 +596,41 @@ Status S3::move_dir(const URI& old_uri, const URI& new_uri) {
   return Status::Ok();
 }
 
+Status S3::copy_file(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  RETURN_NOT_OK(copy_object(old_uri, new_uri));
+  return Status::Ok();
+}
+
+Status S3::copy_dir(const URI& old_uri, const URI& new_uri) {
+  RETURN_NOT_OK(init_client());
+
+  std::string old_uri_string = old_uri.to_string();
+  std::vector<std::string> paths;
+  RETURN_NOT_OK(ls(old_uri, &paths));
+  while (!paths.empty()) {
+    std::string file_name_abs = paths.front();
+    URI file_name_uri = URI(file_name_abs);
+    std::string file_name = file_name_abs.substr(old_uri_string.length());
+    paths.erase(paths.begin());
+    bool dir_exists;
```

Comment:
while we're here, let's add a new-line right _before_ this line to make it a little more readable. thanks!